web/src/engine/websites/Tapread.ts

```diff
@@ -0,0 +1,98 @@
+import { Tags } from '../Tags';
+import icon from './Tapread.webp';
+import { Chapter, DecoratableMangaScraper, Manga, Page, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchCSS, FetchJSON, FetchRequest } from './../FetchProvider';
+
+const imgURL = 'https://static.tapread.com';
+
+type APIMangas = {
+    code: number
+    message: string
+    msg: string
+    result: {
+        moreDetailList: {
+            comicId: number,
+            comicName: string
+        }[]
+    }
+};
+
+type APIChapters = {
+    code: number
+    message: string
+    msg: string
+    result: {
+        chapterList: {
+            chapterNo: number,
+            chapterId: number,
+            comicId: number,
+            chapterName: string
+        }[]
+    }
+};
+
+type APIPages = {
+    code: number
+    message: string
+    msg: string
+    result: {
+        imgList: {
+            imgUrl: string,
+            imgWebpUrl: string,
+        }[]
+    }
+};
+
+@Common.ImageAjax()
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('tapread', `Tapread`, 'https://www.tapread.com', Tags.Language.English, Tags.Source.Official, Tags.Media.Manga, Tags.Media.Manhua, Tags.Media.Manhwa);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return /^https?:\/\/www\.tapread\.com\/comic\/detail\//.test(url);
```

Comment:
Should books also be supported?
https://www.tapread.com/book/detail/35

Reply:
For now, HaKuneko only supports /comic/ (mangas). Haruneko port is on par with Hakuneko. 
Books are novels, and afaik there is still no novel support in Haru.

Nothing prevent use to add Novel support in the future.